homeassistant/components/smlight/button.py

```diff
@@ -62,7 +71,18 @@ async def async_setup_entry(
     """Set up SMLIGHT buttons based on a config entry."""
     coordinator = entry.runtime_data.data
 
-    async_add_entities(SmButton(coordinator, button) for button in BUTTONS)
+    entities = [SmButton(coordinator, button) for button in BUTTONS]
+
+    if coordinator.data.info.zb_type == 1:
+        entities.append(SmButton(coordinator, ROUTER))
+    else:
+        entity_registry = er.async_get(hass)
+        if entity_id := entity_registry.async_get_entity_id(
+            BUTTON_DOMAIN, DOMAIN, f"{coordinator.unique_id}-{ROUTER.key}"
+        ):
+            entity_registry.async_remove(entity_id)
```

Comment:
You could also consider adding a listener to the coordinator and check for this at runtime instead of at update. I have something similar in `airgradient` where I check if the communication method is updated and if it's `cloud`, it deletes all control entities

Reply:
I could do it at runtime, however I don't think its a setting that is going to change that frequently.  At least it seems unlikely that a user would be switching back and forth between coordinator and router modes often.

Reply:
on second thoughts, having the entity created dynamically at runtime will make more sense, once we add the ability to switch Zigbee radio modes from within HA. I have implemented this now.